apps/web/src/features/organizations/components/Dashboard/index.tsx

```diff
@@ -40,17 +42,17 @@ const ViewAllLink = ({ url }: { url: LinkProps['href'] }) => {
 const DASHBOARD_LIST_DISPLAY_LIMIT = 5
 
 const OrganizationsDashboard = () => {
-  const safes = useOrgSafes()
+  const { allSafes: safes, error } = useOrgSafes()
   const isUserSignedIn = useAppSelector(isAuthenticated)
   const orgId = useCurrentOrgId()
   const { activeMembers } = useOrgMembers()
 
   const safesToDisplay = safes.slice(0, DASHBOARD_LIST_DISPLAY_LIMIT)
   const membersToDisplay = activeMembers.slice(0, DASHBOARD_LIST_DISPLAY_LIMIT)
 
-  if (!isUserSignedIn) {
-    return <SignedOutState />
-  }
+  if (!isUserSignedIn) return <SignedOutState />
+
+  if (isUnauthorized(error)) return <UnauthorizedState />
```

Comment:
We do this in every page so I was thinking to extract them into a higher order component but the error comes from different RTK queries so its hard to pull this up but it might be ok if RTK manages to handle the caching.

Reply:
Extracted the two conditions into a new component called `AuthState`